corehq/apps/geospatial/reports.py

```diff
@@ -277,6 +278,31 @@ def _aggregate_query(self, query):
             precision = find_precision(query, case_property)
         return apply_geohash_agg(query, case_property, precision)
 
+    @property
+    def features_from_request(self):
+        features = {}
+        features_json = self.request.GET.get('features')
+        if features_json:
+            try:
+                features |= json.loads(features_json)
+            except json.JSONDecodeError:
+                raise ValueError(f'{features_json!r} parameter is not valid JSON')
+        return features
+
+    @property
+    def saved_polygon_from_request(self):
+        features = {}
+        selected_feature_id = self.request.GET.get('selected_feature_id')
+        if selected_feature_id:
+            selected_polygon = GeoPolygon.objects.get(id=selected_feature_id)
+            geo_json_features = selected_polygon.geo_json['features']
+            features = {uuid.uuid4().hex: feature for feature in geo_json_features}
```

Comment:
nit: how is the hex used?

Reply:
I used the hex format to keep the ID format consistent with the dict of features coming from `features_from_request` which also use a hex UUID for its keys. We don't use the IDs/dict keys directly in the back-end code, but good to keep the same for consistency's sake.